app-backend/db/src/main/scala/SceneToLayerDao.scala

```diff
@@ -172,9 +183,26 @@ object SceneToLayerDao
       scenes_stl.project_layer_id = project_layers.id
       """
     for {
-      queryResult <- (select ++ whereAndOpt(filters: _*) ++ orderByF)
-        .query[SceneToLayerWithSceneType]
-        .to[List]
+      queryResult <- {
+        cachingF(cacheKey)(Some(30 minutes)) {
+          (select ++ whereAndOpt(dbQueryFilters: _*) ++ orderByF)
+            .query[SceneToLayerWithSceneType]
+            .to[List]
+        } map { results =>
+          results.filter { stl =>
+            ((stl.dataFootprint, polygonOption) match {
```

Comment:
Filters for polygons and sceneId subset are pushed down to here now

Reply:
how large is the typical mosaic definition? IIRC memcached has a 1mb max size per key, and fetching a thousand scenes in a project could break things here. In fact, with a complex enough footprint and mask polygon (100kb is large, but I'm not sure that we apply simplification to user provided polygons) in the definition, you could break it with just 10 scenes. Assuming, of course, that it doesn't break in a dozen other areas of the app.

Reply:
That's a good point.

I looked at a few mosaic definitions cached from our development environment -- the largest was 4500 bytes for a mosaic definition with a single scene. If that is about what we'd expect then a project layer with 700 scenes would fit in our 3 mb limit now.

Reply:
I'm fine with putting this off until we actually hit it then, since our largest project which is in actual use is a under 200 scenes. We might want to create an issue / epic for large project support just to keep track of what kinds of things would be required when we get around to it later on.